src/tensor/cpu/allocate.rs

```diff
@@ -64,22 +66,19 @@ impl<E: Unit> ZeroFillStorage<E> for Cpu {
     }
 }
 
-impl OnesTensor<f32> for Cpu {
-    fn try_ones_like<S: HasShape>(
-        &self,
-        src: &S,
-    ) -> Result<Tensor<S::Shape, f32, Self>, Self::Err> {
-        let storage = StridedArray::try_new_with(*src.shape(), 1.0)?;
+impl<E: Unit + One> OnesTensor<E> for Cpu {
```

Comment:
@nkoppel FYI related to your One change - we could pull out the num_traits dependency into a smaller PR that just adds it for this for now (probably best to stage this giant PR anyway).

Reply:
Unfortunately, One is not implemented for bool, so my current implementation is probably the best we can do.